CascadeEnergy/good-graylog2
# SEN-1558 -- Good Reporter for Graylog2

## Graylog Setup:

For the graylog server you are logging to: Ensure there is a _GELF_UDP_ input configured.
## Hapi Setup:

Use this in the same manner as `good-console`. The `config` passed into the reporter look like: 

```
var config = {
  host: <graylog-host>
  port: <gelf-udp-input-port>
  container: <docker-container-id> //optional
}
```
### To Test

Build a graylog box with the instructions provided here: http://docs.graylog.org/en/1.1/pages/installation/aws.html

Put that box in a federation with your "pym"/"dev" box, and make sure you have ssh access via a security group as well.
As described in the graylog instructions, ssh into the graylog box once it's done building and run `sudo graylog-ctl reconfigure`

Open the url to the graylog server. Login as admin/admin
Go to System -> Inputs
Create a GELF UDP input, give it a name "gelfudp" and submit it.
Once it's created hit "show received messages"

ssh into your pym box.

```
cd /var/cascade/
mkdir good-graylog2-test
cd good-graylog2-test
npm install nwarlen/good-graylog2#SEN-1558
npm install good hapi
vim server.js
```

paste this into it. And then replace the "host" config with the Private IP of your graylog instance.

``` javascript
'use strict';

var Hapi = require('hapi');
var good = require('good');
var graylogReporter = require('good-graylog2');

var server = new Hapi.Server();

server.connection({
  port: 1234,
  routes: {cors: true}
});


server.route([
  {
    method: 'GET',
    path: '/healthcheck',
    handler: function(request, reply) {
      reply('OK');
    }
  }
]);

var goodOptions = {
  reporters: [
    {
      reporter: graylogReporter,
      events: {
        response: '*',
        log: '*',
        error: '*'
      },
      config: {
        service: 'service-name',
        host: '172.21.78.148',
        port: '12201',
      }
    }
  ],
  responsePayload: true
};

server.register({ register: good, options: goodOptions }, function(err) {
  if(err) {
    server.log(['service-name', 'error'], err);
  } else {
    server.start(function() {
      server.log(['service-name', 'info'], 'service-name running at: ' + server.info.uri);
    });
  }
});
```

```
node server.js
```

Refresh the search in your browser window of graylog. Should see the server started log.
